samples/quickperf/src/main/java/com/google/cloud/jdbc/quickperf/QuickPerf.java

```diff
@@ -0,0 +1,497 @@
+            boolean isGoogleSQL;
+            if (connection.unwrap(CloudSpannerJdbcConnection.class).getDialect()
+                    .equals(Dialect.GOOGLE_STANDARD_SQL)) {
+                isGoogleSQL = true;
+            } else {
+                isGoogleSQL = false;
+            }
+
+            connection.setAutoCommit(false);
+
+            // if there is DML switch to r/w transaction mode and apply transaction tagging.
+            // Otherwise set to read-only mode.
+            if (config.getQuery().contains("INSERT") ||
+                    config.getQuery().contains("UPDATE")
+                    || config.getQuery().contains("DELETE")) {
+                // read-write
+                connection.createStatement().execute("SET TRANSACTION READ WRITE");
+
+                if (isGoogleSQL) {
+                    connection.createStatement().execute(String.format("SET TRANSACTION_TAG = '%s'", DEFAULT_TAG));
+                } else {
+                    connection.createStatement()
+                            .execute(String.format("SET SPANNER.TRANSACTION_TAG = '%s'", DEFAULT_TAG));
+                }
+
+            } else {
+                // read-only
+                // connection.createStatement().execute("SET TRANSACTION READ ONLY");
+                connection.setAutoCommit(true);
+            }
+
+            PreparedStatement statement = connection.prepareStatement(config.getQuery());
+            int batchCounter = config.getBatchSize();
+            int batchRound = 0;
+
+            for (int i = 0; i < config.getIterations(); i++) {
+                if (config.getBatchSize() == 0) {
+                    // single statements
+                    try {
+                        if (config.getQueryParams() != null) {
+                            statement = parametrizeStatement(statement, config.getQueryParams());
+                        }
+
+                        if (isGoogleSQL) {
+                            connection.createStatement().execute(String.format("SET STATEMENT_TAG='%s'",
+                                    DEFAULT_TAG));
+                        } else {
+                            connection.createStatement().execute(String.format("SET SPANNER.STATEMENT_TAG='%s'",
+                                    DEFAULT_TAG));
+                        }
+
+                        boolean hasResults = false;
+                        long start = 0;
+                        long stop = 0;
+                        if (connection.getAutoCommit()) {
+                            // read-only
+                            start = System.nanoTime();
+                            hasResults = statement.execute();
+                            stop = System.nanoTime() - start;
+                        } else {
+                            start = System.nanoTime();
+                            hasResults = statement.execute();
+                            connection.commit();
```

Comment:
Is there a specific reason that you want to measure it in this way with a separate `execute` and `commit` call? Would it not make more sense to just use auto-commit for DML without batching as well?